.github/workflows/publish-dist.yml

```diff
@@ -23,40 +24,37 @@ jobs:
           git config --global credential.helper store
           echo 'https://token:${{ secrets.COCKPITUOUS_TOKEN }}@github.com' >> ~/.git-credentials
 
-      - name: Commit dist tarball dist repo
+      - name: Commit to dist repo
         run: |
           set -ex
-
-          # we need a fully predictable name/URL, and git introduces these
-          # additional numbers into the version; so wrap it in a predictable named tar
-          sha=$(ls -d dist-* | sed 's/dist-//')
+          SHA=$(cat SHA)
 
           # multiple parallel workflows race against each other, so the final
           # `git push` may fail
           rc=1
           for retry in $(seq 5); do
```

Comment:
I don't like this loop, and I think we should be doing this differently:

 - on pull requests, we should push only a tag, as a commit on top of the most recent commit on the main branch.
 - on merge to main, we then push an update also to the main branch here

random PRs ending up on the main branch seems weird, and is also exactly the source of the potential conflict here.

thoughts?

Reply:
We quickly discussed that on IRC. Alternatives are to just push "bare" tags which are not attached to a branch, or push every SHA build as its own branch. The latter would also make it possible to do `git clone -b ` directly, but I suppose it's somewhat more expensive to store in git (precisely *because* `git clone -b` is possible, so it must publish more metadata). However, these would all lose the nice feature of delta compression against the previous version, so that would not really be an improvement.

Reply:
I generally like your idea about pushing to main only for main builds. However, wouldn't we still need the same retry loop? If you land two PRs closely together, one of their publish-dist runs will still lose the race and needs to be retried. It would just be less likely.